tests/examples/wibox/template.lua

```diff
@@ -0,0 +1,63 @@
+local file_path, image_path, luacovpath = ...   
+
+-- Set the global shims
+-- luacheck: globals awesome
+awesome = require( "awesome" )
+-- luacheck: globals client
+client  = require( "client"  )
+-- luacheck: globals tag
+tag     = require( "tag"     )
+
+-- Force luacheck to be silent about setting those as unused globals
+assert(awesome and client and tag)
+
+local wibox   = require( "wibox"         )
+local GLib    = require( "lgi"           ).require("GLib")
+local surface = require( "gears.surface" )
+
+-- If luacov is available, use it. Else, do nothing.
+pcall(function()
+    require("luacov.runner")(luacovpath)
+end)
+
+-- Unless there is some kind of shim for that, the Awesome mainloop need to be
+-- emulated.
+local main_loop = GLib.MainLoop()
+
+-- This is the main widget the tests will use as top level
+local container = wibox.layout.fixed.vertical()
+
+-- Let the test request a size and file format
+local w, h, image_type = loadfile(file_path)(container)
+image_type = image_type or "svg"
+
+-- Emulate 10 cycles of mainloop
+local counter = 1
+
+-- Loop as fast as possible
+GLib.timeout_add_seconds(0, 0,function()
+    -- Emulate capi.awesome mainloop signal. Otherwise gears.timer wont work
+    awesome:emit_signal("refresh")
+
+    counter = counter + 1
+    if counter < 10 then
+        return true
+    else
+        -- Get the example fallback size (the tests can return a size if the want)
+        local f_w, f_h = container:fit({dpi=96}, 9999, 9999)
+
+        -- There is an overhead that cause testbox "...", add 10 until someone
+        -- figures out the real equation
+        f_w, f_h = f_w+10, f_h+10
+
+        -- Save to the output file
+        surface["widget_to_"..image_type](container, image_path.."."..image_type, w or f_w, h or f_h)
```

Comment:
There is the png version too, it's not in this PR, but it will come when I will need it. This PR is already 1k line with only the code I need...

(there is no huge second part, don't worry ;) ).